api/src/pcapi/core/providers/titelive_apply_new_filters_on_history.py

```diff
@@ -0,0 +1,344 @@
+
+HEADER_LINE_COUNT = 1
+COMMENT_COLUMN_INDICES = {
+    "EAN": 0,
+    "COMMENT": 1,
+}
+COLUMN_INDICES = {
+    "EAN": 0,
+    "CODE_SUPPORT": 1,
+    "IDLECTORAT": 2,
+    "TAUX_TVA": 3,
+    "CODE_GTL": 4,
+    "TITLE": 7,
+}
+
+
+# This if the exact copy of offers_api.reject_inappropriate_products with dry run support and return list of email instead of bool
+def reject_inappropriate_products(ean: str, dry: bool = False) -> list[str]:
+    products = models.Product.query.filter(models.Product.extraData["ean"].astext == ean).all()
+    emails: list[str] = []
+
+    if not products:
+        return emails
+
+    for product in products:
+        product.isGcuCompatible = False
+        db.session.add(product)
+
+    offers_query = models.Offer.query.filter(
+        models.Offer.productId.in_(p.id for p in products),
+        models.Offer.validation != models.OfferValidationStatus.REJECTED,
+    ).options(sa.orm.joinedload(models.Offer.stocks).joinedload(models.Stock.bookings))
+
+    offers = offers_query.all()
+
+    if not dry:
+        offers_query.update(
+            values={
+                "validation": models.OfferValidationStatus.REJECTED,
+                "lastValidationDate": datetime.datetime.utcnow(),
+                "lastValidationType": OfferValidationType.CGU_INCOMPATIBLE_PRODUCT,
+            },
+            synchronize_session="fetch",
+        )
+
+    offer_ids = []
+    for offer in offers:
+        offer_ids.append(offer.id)
+        if dry:
+            bookings = list(itertools.chain.from_iterable(stock.bookings for stock in offer.stocks))
+        else:
+            bookings = bookings_api.cancel_bookings_from_rejected_offer(offer)
+
+        for booking in bookings:
+            emails.append(booking.user.email)
+
+        # `send_booking_cancellation_emails_to_user_and_offerer` would send an e-mail to the pro and
+        # none to the beneficiary. We want the opposite. E-mails will be sent by a separate script.
+        # if send_booking_cancellation_emails and not dry:
+        #     for booking in bookings:
+        #         send_booking_cancellation_emails_to_user_and_offerer(booking, reason=BookingCancellationReasons.FRAUD)
+
+    try:
+        if not dry:
+            db.session.commit()
+    except Exception as exception:  # pylint: disable=broad-except
+        logger.exception(
+            "Could not mark product and offers as inappropriate: %s",
+            extra={"ean": ean, "products": [p.id for p in products], "exc": str(exception)},
+        )
+        return emails
+    logger.info(
+        "Rejected inappropriate products",
+        extra={"ean": ean, "products": [p.id for p in products], "offers": offer_ids},
+    )
+
+    if offer_ids and not dry:
+        favorites = users_models.Favorite.query.filter(users_models.Favorite.offerId.in_(offer_ids)).all()
+        repository.delete(*favorites)
+        search.async_index_offer_ids(offer_ids)
```

Comment:
J'ai oublié ça : sur staging, on a un quota limité d'opérations vers l'API d'Algolia. Donc si on a 1.000.000 d'offres à réindexer, mieux vaudrait conditionner sur staging, par exemple avec : `if not settings.IS_STAGING`.

Reply:
Merci, je rajoute